usr/share/rear/build/GNU/Linux/400_copy_modules.sh

```diff
@@ -1,28 +1,129 @@
 # 400_copy_modules.sh
 #
-# copy kernel modules for Relax-and-Recover
+# Copy kernel modules to the rescue/recovery system.
+
 #
 # This file is part of Relax-and-Recover, licensed under the GNU General
 # Public License. Refer to the included COPYING for full text of license.
 
-Log "Collecting modules for kernel version $KERNEL_VERSION"
-MODFILES=(
-$( ResolveModules "$KERNEL_VERSION" "${MODULES[@]}" "${MODULES_LOAD[@]}"  )
-)
-StopIfError "Could not resolve kernel module dependancies"
+# The special value MODULES=( 'none' ) enforces that
+# no kernel modules get included in the rescue/recovery system
+# regardless of what modules are currently loaded.
+# Test the first MODULES array element because other scripts
+# in particular rescue/GNU/Linux/240_kernel_modules.sh
+# already appended other modules to the MODULES array:
+if test "none" = "$MODULES" ; then
+    LogPrint "Omit copying kernel modules (MODULES contains 'none')"
+    return
+fi
+
+# Artificial 'for' clause that is run only once
+# to be able to 'continue' with the code after it
+# when the kernel modules have been copied into the rescue/recovery system
+# (the 'for' loop is run only once so that 'continue' is the same as 'break')
+# which avoids dowdy looking code with deeply nested 'if...else' conditions:
+for dummy in "once" ; do
+
+    # The special user setting MODULES=( 'all' ) enforces that
+    # all files from the /lib/modules/* directories
+    # get included in the rescue/recovery system
+    # regardless of what is set in EXCLUDE_MODULES.
+    # Test all MODULES array members to make the 'all' functionality work for
+    # MODULES array contents like MODULES=( 'moduleX' 'all' 'moduleY' ):
+    if IsInArray "all" "${MODULES[@]}" ; then
+        LogPrint "Copying all kernel modules in /lib/modules/* (MODULES contains 'all')"
+        # The '--parents' is needed to get the '/lib/' directory in the copy:
+        if ! cp $verbose -t $ROOTFS_DIR -a --parents /lib/modules 1>&2 ; then
```

Comment:
How about restricting that to the kernel version that is used? Why should we copy all the modules from all the other potentially installed kernels?